lib/galaxy/config/__init__.py

```diff
@@ -258,36 +250,37 @@ def _process_config(self, kwargs):
             self.thread_local_log = threading.local()
 
         # Install database related configuration (if different).
-        self.install_database_connection = kwargs.get("install_database_connection", None)
         self.install_database_engine_options = get_database_engine_options(kwargs, model_prefix="install_")
 
         # Where dataset files are stored
-        self.file_path = resolve_path(kwargs.get("file_path", "files"), self.data_dir)
+        self.file_path = os.path.join(self.data_dir, self.file_path)
         # new_file_path and legacy_home_dir can be overridden per destination in job_conf.
-        self.new_file_path = resolve_path(kwargs.get("new_file_path", "tmp"), self.data_dir)
+        self.new_file_path = os.path.join(self.data_dir, self.new_file_path)
         override_tempdir = string_as_bool(kwargs.get("override_tempdir", "True"))
         if override_tempdir:
             tempfile.tempdir = self.new_file_path
         self.shared_home_dir = kwargs.get("shared_home_dir", None)
-        self.openid_consumer_cache_path = resolve_path(kwargs.get("openid_consumer_cache_path", "openid_consumer_cache"), self.data_dir)
+        self.openid_consumer_cache_path = os.path.join(self.data_dir, self.openid_consumer_cache_path)
         self.cookie_path = kwargs.get("cookie_path", None)
-        self.tool_path = resolve_path(kwargs.get("tool_path", "tools"), self.root)
-        self.tool_data_path = resolve_path(kwargs.get("tool_data_path", "tool-data"), os.getcwd())
+        self.tool_path = os.path.join(self.root, self.tool_path)
+        self.tool_data_path = os.path.join(self.root, self.tool_data_path)
         if not running_from_source and kwargs.get("tool_data_path", None) is None:
-            self.tool_data_path = resolve_path("tool-data", self.data_dir)
-        self.builds_file_path = resolve_path(kwargs.get("builds_file_path", os.path.join(self.tool_data_path, 'shared', 'ucsc', 'builds.txt')), self.root)
-        self.len_file_path = resolve_path(kwargs.get("len_file_path", os.path.join(self.tool_data_path, 'shared', 'ucsc', 'chrom')), self.root)
+            self.tool_data_path = os.path.join(self.data_dir, "tool-data")
+        if self.builds_file_path is None:
```

Comment:
```suggestion
        self.builds_file_path = os.path.join(self.tool_data_path, self.builds_file_path)
```

Reply:
I'll commit this suggestion, but here's my reasoning for testing for None.
Currently, this cannot be None. However, if it is, `os.path.join` will, of course, complain. And here's how it could become not None: although a default value is currently assigned, we should be able to override that with a null (see https://github.com/galaxyproject/galaxy/pull/8648#issuecomment-532300644 + preseeding comment by @jmchilton).  I will implement this in a follow-up PR. Also, once we iron out #8651, most paths requiring resolving will be handled differently. The old version had the default hardcoded in this file, so it was safe. I'm OK not checking for None (for now), as this will be rewritten very soon. (Please correct me if this reasoning is incorrect.)

Reply:
I don't get what's the use case for allowing these to be `None`, to be honest. I'd not go down that route if it's not necessary.